src/image/ExposureFitsReader.cc

```diff
@@ -0,0 +1,358 @@
+/*
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "lsst/log/Log.h"
+
+#include "lsst/afw/image/Calib.h"
+#include "lsst/afw/geom/polygon/Polygon.h"
+#include "lsst/afw/geom/SkyWcs.h"
+#include "lsst/afw/image/ApCorrMap.h"
+#include "lsst/afw/detection/Psf.h"
+#include "lsst/afw/image/TransmissionCurve.h"
+#include "lsst/afw/image/ExposureFitsReader.h"
+
+namespace lsst { namespace afw { namespace image {
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("afw.image.fits.ExposureFitsReader");
+
+}  // anonymous
+
+class ExposureFitsReader::MetadataReader {
+public:
+
+    MetadataReader(
+        std::shared_ptr<daf::base::PropertyList> primaryMetadata,
+        std::shared_ptr<daf::base::PropertyList> imageMetadata
+    ) {
+        // Try to read WCS from image metadata, and if found, strip the keywords used
+        try {
+            wcs = afw::geom::makeSkyWcs(*imageMetadata, true);
+        } catch (lsst::pex::exceptions::TypeError) {
+            LOGLS_DEBUG(_log, "No WCS found in FITS metadata");
```

Comment:
Yes, it will leave it empty or null (the distinction, to the extent that there is one, shouldn't matter in this context).  We are careful not to dereference it - I believe the `if` block immediately below is the only place where `ExposureFitsReader` dereferences `wcs` itself, and that `if` block checks first that the pointer is non-null.